### ISSUE TITLE
gRPC service throw warning when using @Blocking and OpenTelemetry gRPC interceptor

### ISSUE THREAD

**Issue by XSAM** [quarkusio/quarkus#24395]
### Describe the bug

Calling a gRPC method with @Blocking annotation with gRPC interceptor produces warnings when enabling [io.opentelemetry.instrumentation:opentelemetry-grpc-1.6:1.10.0-alpha](https://mvnrepository.com/artifact/io.opentelemetry.instrumentation/opentelemetry-grpc-1.6/1.10.0-alpha) and io.quarkus:quarkus-opentelemetry-exporter-otlp.

It does not affect gRPC methods without @Blocking annotation.

### Expected behavior

no warning

### Actual behavior

a warning `Context in storage not the expected context, Scope.close was not called correctly`

### How to Reproduce?

1. Clone https://github.com/XSAM/quarkus-issue-24395-reproducer
2. Checkout https://github.com/XSAM/quarkus-issue-24395-reproducer/commit/a28bc52d28720c89303e2bce4b52ca848bb6c74f
3. Run `mvn test` to see the warning

### Output of `uname -a` or `ver`

21.2.0

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 OpenJDK Runtime Environment (build 17.0.2+8-86) OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)

### Quarkus version or git rev

2.7.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.3.3

### Additional information

Related to https://github.com/quarkusio/quarkus/issues/23805

**Comment by radcortez** [comment]
The gRPC OTel library does not work out of the box, because the interceptors need to be integrated manually. I've added support for gRPC directly into Quarkus here: https://github.com/quarkusio/quarkus/pull/24117

Are you able to try the main version? Thanks!

**Comment by cescoffier** [comment]
@radcortez why did you re-open the issue? Shouldn't it be closed as you added the support?

**Comment by radcortez** [comment]
This seems to be the new issue following your comment in: https://github.com/quarkusio/quarkus/issues/23805#issuecomment-1071210302.

But the reproducer is still using the native OTel gRPC library and not our implementation, so I've suggested updating the reproducer to use our version. I've closed the issue by accident.

**Comment by XSAM** [comment]
I updated the reproducer, and it uses quarkus implementation now. See https://github.com/XSAM/quarkus-issue-24395-reproducer/commit/4d73c84785571e2217f2751f738eeacd69eb8a1b. (It now uses the Quarkus local build)

The warning message is gone, but I found other issues:

- Cannot see client spans. The TracingClientInterceptor seems not working for the blocking or non-blocking gRPC method.
- Cannot get Context inside the blocking gRPC service's handler. (I have tried this commit https://github.com/quarkusio/quarkus/commit/cc89f6cbdcbaf7f2f2e486eb305a77bfd3189a9f) Related issue: https://github.com/quarkusio/quarkus/issues/26222

**Comment by radcortez** [comment]
There is an issue with blocking services where the context is not propagated, fixed by #27100.

For client spans, if you create the client manually, interceptors are not automatically added. You should inject the client managed by Quarkus:

```java
@QuarkusTest
class HelloWorldServiceTest {
    @GrpcClient
    GreeterGrpc.GreeterBlockingStub client;
    @Inject
    InMemorySpanExporter inMemorySpanExporter;

    @Test
    public void testHelloWorldServiceUsingBlockingStub() {
        HelloReply reply = client.sayHello(HelloRequest.newBuilder().setName("neo-blocking").build());
        assertThat(reply.getMessage()).isEqualTo("Hello neo-blocking");

        await().atMost(5, TimeUnit.SECONDS).until(() -> inMemorySpanExporter.getFinishedSpanItems().size() == 2);

        List<SpanData> spans = inMemorySpanExporter.getFinishedSpanItems();
        SpanData server = spans.get(0);
        SpanData client = spans.get(1);

        assertEquals(server.getTraceId(), client.getTraceId());

        assertEquals(SpanKind.SERVER, server.getKind());
        assertEquals(SpanKind.CLIENT, client.getKind());
    }

    @ApplicationScoped
    static class InMemorySpanExporterProducer {
        @Produces
        @Singleton
        InMemorySpanExporter inMemorySpanExporter() {
            return InMemorySpanExporter.create();
        }
    }
}
```
